reconcile/dynatrace_token_provider/validate.py

```diff
@@ -0,0 +1,60 @@
+from collections.abc import Iterable
+
+from reconcile.gql_definitions.dynatrace_token_provider.token_specs import (
+    DynatraceTokenProviderTokenSpecV1,
+)
+
+
+class SecretNotUniqueError(Exception):
+    pass
+
+
+class TokenNameNotUniqueInSecretError(Exception):
+    pass
+
+
+class KeyNameNotUniqueInSecretError(Exception):
+    pass
+
+
+class TokenNameTooLongError(Exception):
+    pass
+
+
+MAX_TOKEN_NAME_LENGTH = 46
+
+
+def validate_token_specs(specs: Iterable[DynatraceTokenProviderTokenSpecV1]) -> None:
+    """
+    We cannot catch all potential errors through json schema definition.
+    """
+    for spec in specs:
+        seen_secrets: set[str] = set()
+        for secret in spec.secrets:
+            secret_definition_key = f"{secret.namespace}/{secret.name}"
+            if secret_definition_key in seen_secrets:
+                raise SecretNotUniqueError(
+                    f"A secret cannot be re-defined and must be unique per spec. Secret '{secret.name}' in namespace '{secret.namespace}' is defined multiple times in spec '{spec.name}'."
+                )
+            seen_secrets.add(secret_definition_key)
+
+            seen_tokens: set[str] = set()
+            seen_key_names: set[str] = set()
+            for token in secret.tokens:
+                if len(token.name) > MAX_TOKEN_NAME_LENGTH:
+                    raise TokenNameTooLongError(
+                        f"Token name '{token.name}' is too long. The maximum length is {MAX_TOKEN_NAME_LENGTH} characters."
+                    )
+
+                if token.name in seen_tokens:
+                    raise TokenNameNotUniqueInSecretError(
+                        f"A token name must be unique within a secret. Token name '{token.name}' is used more than once in secret '{secret.name}' in token spec '{spec.name}'."
+                    )
+                seen_tokens.add(token.name)
+
+                if token.key_name_in_secret:
+                    if token.key_name_in_secret in seen_key_names:
+                        raise KeyNameNotUniqueInSecretError(
+                            f"A key name must be unique within a secret. Key name '{token.key_name_in_secret}' is used more than once in secret '{secret.name}' in token spec '{spec.name}'."
+                        )
+                    seen_key_names.add(token.key_name_in_secret)
```

Comment:
`token.name` and `token.key_name_in_secret` used as `secret_key = token.key_name_in_secret or token.name`, should apply combined unique to `secret_key`? https://github.com/app-sre/qontract-reconcile/blob/e41d5c27eec7fc8a181b1d6636d25cfa7c0d891f/reconcile/dynatrace_token_provider/integration.py#L381

Reply:
Awesome catch! Added a new test case for this scenario

Reply:
Is there a use case for `token.name` to be unique? Or just `secret_key` unique is enough?

Reply:
Yes `token.name` is used for name in DT API and should be unique.